states.js

```diff
@@ -31,61 +54,112 @@ class ConditionOnsetState extends BaseState {
         this.target_encounter = target_encounter;
         this.codes = codes;
     }
+    toJSON() {
+        return {
+            ...super.toJSON(),
+            "target_encounter":this.target_encounter,
+            "codes":this.codes
+        }
+    }
 }
 class ConditionEndState extends BaseState {
     constructor(name) {
         super(name);
         this.type = 'ConditionEnd';
     }
+    toJSON() {
+        return {
+            ...super.toJSON()
+        }
+    }
 }
-class AlleryOnsetState extends BaseState {
+class AllergyOnsetState extends BaseState {
     constructor(name, target_encounter, codes) {
         super(name);
         this.type = 'AllergyOnset';
         this.target_encounter = target_encounter;
         this.codes = codes;
     }
+    toJSON() {
+        return {
+            ...super.toJSON(),
+            "target_encounter":this.target_encounter,
+            "codes":this.codes
+        }
+    }
 }
 class AllergyEndState extends BaseState {
     constructor(name) {
         super(name);
         this.type = 'AllergyEnd';
     }
+    toJSON() {
```

Comment:
I should have mentioned this before, but you shouldn't need to implement `toJSON` in states where you don't need anything other than the BaseState fields (aka, if all you're doing is calling `...super.toJSON()`). If you remove these, it should just call the `BaseState`'s version of `toJSON()`.